core/metadata_base.h

```diff
@@ -53,6 +66,17 @@
 #define FILENAME_XMIPP_SQLITE "SQLite format 3"
 #define DEFAULT_BLOCK_NAME "noname"
 
+/* ITERATING OVER METADATA
+ * You can iterate directly over abstract MetaData as well as over specific MetaData*.
+ * You can iterate over:
+ *  1) Ids of rows (arbitrary size_t): for (size_t id : md.ids())
+ *  2) Rows: for (const MDRow& row : md)
+ * Never rely on any properties of IDs! E.g. they are ascending, positive, continuous etc.
+ * There is a hierarchy of MDRow classes copying MetaData hierarchy: MDRow, MDRowVec, MDRowSql.
+ * If you need row-specific function, do a dynamic cast:
```

Comment:
This example is bad! If `MetaData` is `MetaDataVec`, one should use `MetaDataVec` type in function signature to avoid as many casting as possible. This helps to catch all possible errors as compile-time.